src/Microsoft.Owin.Security.WsFederation/WsFederationAuthenticationHandler.cs

```diff
@@ -299,8 +298,23 @@ public override Task<bool> InvokeAsync()
                 tvp.ValidIssuers = (tvp.ValidIssuers == null ? issuers : tvp.ValidIssuers.Concat(issuers));
                 tvp.IssuerSigningKeys = (tvp.IssuerSigningKeys == null ? _configuration.SigningKeys : tvp.IssuerSigningKeys.Concat(_configuration.SigningKeys));
 
-                SecurityToken parsedToken;
-                ClaimsPrincipal principal = Options.SecurityTokenHandlers.ValidateToken(token, tvp, out parsedToken);
+                ClaimsPrincipal principal = null;
+                SecurityToken parsedToken = null;
+                foreach (var validator in Options.SecurityTokenHandlers)
+                {
+                    if (validator.CanReadToken(token))
+                    {
+                        principal = validator.ValidateToken(token, tvp, out parsedToken);
+                        break;
+                    }
+                }
+
+                // what to do here?
+                if (principal == null)
```

Comment:
it could be the case the validator returned null OR no validator could read the token.
I agree in either case, you need to throw.